### PR TITLE
fix: resolve integration test failures by migrating COCO image loading to HF Hub fixtures

### LINKED ISSUE #47205: Standardize test image fixture loading for COCO images
### Problem

Several tests still download COCO fixture images directly with `httpx.get(...)` or `requests.get(...)`. This bypasses `utils/fetch_hub_objects_for_ci.py` and the existing `url_to_local_path(...)` fallback behavior, so CI prefetching can succeed while local or partial test runs still hit `http://images.cocodataset.org/...` directly.

We should have one test helper for loading image fixtures from URLs, and use it instead of open-coded `httpx` / `requests` calls in tests.

### Proposed change

Add a small helper in the shared test utilities, likely next to `url_to_local_path` in `tests/test_processing_common.py`, for example:

```python
def load_test_image(url: str):
    return load_image(url_to_local_path(url))
```

Then update tests that currently do this:

```python
Image.open(io.BytesIO(httpx.get(url, follow_redirects=True).content))
Image.open(requests.get(url, stream=True).raw)
```

to use the shared helper:

```python
image = load_test_image(url)
```

For COCO fixtures, the URLs should point at the Hub mirror under `hf-internal-testing/fixtures-coco`, not the original `images.cocodataset.org` host. The matching entries in `utils/fetch_hub_objects_for_ci.py` should also stay aligned with whatever URLs the tests use.

### Direct `httpx.get` / `requests.get` COCO call sites to update

`httpx.get(...)`:

- `tests/test_image_processing_common.py:180`
- `tests/models/vitpose/test_image_processing_vitpose.py:238`
- `tests/models/flava/test_image_processing_flava.py:409`
- `tests/models/pix2struct/test_image_processing_pix2struct.py:109`
- `tests/models/deepseek_vl_hybrid/test_image_processing_deepseek_vl_hybrid.py:212`
- `tests/models/kimi_k25/test_image_processing_kimi_k25.py:272`
- `tests/models/idefics/test_image_processing_idefics.py:156`
- `tests/models/fuyu/test_image_processing_fuyu.py:185`
- `tests/models/qwen2_vl/test_image_processing_qwen2_vl.py:299`

`requests.get(...)`:

- `tests/models/aimv2/test_modeling_aimv2.py:482`
- `tests/models/aimv2/test_modeling_aimv2.py:516`
- `tests/models/aimv2/test_modeling_aimv2.py:547`
- `tests/models/video_llama_3/test_image_processing_video_llama_3.py:300`
- `tests/models/eomt/test_modeling_eomt.py:178`
- `tests/models/eomt/test_modeling_eomt.py:221`
- `tests/models/eomt/test_modeling_eomt.py:237`
- `tests/models/eomt/test_modeling_eomt.py:274`
- `tests/models/eomt/test_modeling_eomt.py:323`
- `tests/models/eomt/test_modeling_eomt.py:369`
- `tests/models/eomt_dinov3/test_modeling_eomt_dinov3.py:223`
- `tests/models/eomt_dinov3/test_modeling_eomt_dinov3.py:265`
- `tests/models/eomt_dinov3/test_modeling_eomt_dinov3.py:322`
- `tests/models/eomt_dinov3/test_modeling_eomt_dinov3.py:378`
- `tests/models/eomt_dinov3/test_modeling_eomt_dinov3.py:424`
- `tests/models/eomt_dinov3/test_modeling_eomt_dinov3.py:466`
- `tests/models/minimax_m3_vl/test_modeling_minimax_m3_vl.py:700`
- `tests/models/layoutlmv2/test_image_processing_layoutlmv2.py:152`
- `tests/models/lfm2_vl/test_modeling_lfm2_vl.py:214`
- `tests/models/lfm2_vl/test_modeling_lfm2_vl.py:300`
- `tests/models/aria/test_modeling_aria.py:324`
- `tests/models/aria/test_modeling_aria.py:362`
- `tests/models/aria/test_modeling_aria.py:402`
- `tests/models/imagegpt/test_image_processing_imagegpt.py:289`
- `tests/models/llava/test_modeling_llava.py:370`
- `tests/models/llava/test_modeling_llava.py:421`
- `tests/models/llava/test_modeling_llava.py:474`
- `tests/models/fast_vlm/test_modeling_fast_vlm.py:285`

### Follow-up scope

There are more COCO URL literals in pipeline tests and model tests that are passed directly to pipelines/processors rather than downloaded with `httpx` or `requests`. Those can be handled separately by standardizing URL constants and wrapping them with `url_to_local_path(...)` where the test accepts local paths.

### PR DESCRIPTION
**Pull request by abhishekkapoorx**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47219)

# What does this PR do?

This PR resolves the integration test failures triaged in #47205. 

The domain `images.cocodataset.org` has been unstable, resulting in connection timeouts and failing integration tests across multiple model pipelines. To address this, we have migrated COCO-dataset image loading in our test suites to use the mirrored fixtures on the Hugging Face Hub under `hf-internal-testing/fixtures-coco` and utilizing local path redirection.

### Key Changes:
1. **Added test helper**: Introduced `load_test_image` in `tests/test_processing_common.py` which wraps `load_image` and automatically processes remote URLs with `url_to_local_path`.
2. **Refactored tests**: Converted occurrences of direct `requests.get` or `httpx.get` targeting `http://images.cocodataset.org/val2017/000000039769.jpg` to the new `load_test_image` helper pointing to the Hugging Face mirror.
3. **Cleaned up CI downloads**: Removed direct references to stale `images.cocodataset.org` URLs from the `URLS_FOR_TESTING_DATA` registry in `utils/fetch_hub_objects_for_ci.py`.

This prevents flaky network issues and leverages local caching during CI execution.

## Related Issues
Fixes #47205

## Code Agent Policy

The Transformers repo is currently being overwhelmed by a large number of PRs and issue comments written by
code agents. These often are low-quality, or fix extremely minor issues that occur rarely or never in practice.
As a result, we're instituting a rule that **first-time contributors should not use code agents to submit PRs or issues**.
We'd also ask autonomous "OpenClaw"-like agents not to open any PRs or issues.

Issues/PRs from first-time contributors that violate this rule will probably just be closed without review, and we
might block you, especially if you open more than one or appear to be deliberately ignoring this. We especially do not
want new contributors to jump in on random issues to contribute an agent-written fix. This creates lots of noise
for reviewers and other users and will almost certainly get you blocked.

For more information, please read [`CONTRIBUTING.md`](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md).

- [ ] (First-time contributors only): I confirm that this PR description and code is not written by an LLM or code agent

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://huggingface.co/docs/transformers/contributing) and the
      [Pull Request](https://huggingface.co/docs/transformers/pr_checks) checks?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes according to the [guidelines](https://github.com/huggingface/transformers/tree/main/docs)?
- [ ] Did you write any new necessary [tests](https://huggingface.co/docs/transformers/testing)?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/models/layoutlmv2/test_image_processing_layoutlmv2.py [hunk 1]
```diff
@@ -31,13 +30,14 @@
...
 from ...test_image_processing_common import ImageProcessingTestMixin, prepare_image_inputs
+from ...test_processing_common import load_test_image
 
 
 if is_torch_available():
     import torch
 
 if is_pytesseract_available():
-    from PIL import Image
+    pass
```
Comment (ydshieh) [inline tests/models/layoutlmv2/test_image_processing_layoutlmv2.py]: if we no longer need it , simply remove it

File: tests/models/eomt_dinov3/test_modeling_eomt_dinov3.py [hunk 2]
```diff
@@ -31,14 +29,15 @@
...
+from ...test_processing_common import load_test_image
 
 
 if is_torch_available():
     import torch
 
 
 if is_vision_available():
-    from PIL import Image
+    pass
```
Comment (ydshieh) [inline tests/models/eomt_dinov3/test_modeling_eomt_dinov3.py]: just remove it

File: tests/models/eomt/test_modeling_eomt.py [hunk 3]
```diff
@@ -15,23 +15,22 @@
...
+from ...test_processing_common import load_test_image
 
 
 if is_torch_available():
     import torch
 
 
 if is_vision_available():
-    from PIL import Image
+    pass
```
Comment (ydshieh) [inline tests/models/eomt/test_modeling_eomt.py]: remove the block

File: tests/models/layoutlmv2/test_image_processing_layoutlmv2.py (hunk 1 above)
Comment (abhishekkapoorx) [reply tests/models/layoutlmv2/test_image_processing_layoutlmv2.py]: Sure I'll remove the blocks.

**Comment by abhishekkapoorx** [comment]
@ydshieh In [src/transformers/modeling_utils.py](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py), we are using the `axis` but latest documentation of torch uses `dim`. I think we should fix this as well.
```diff --git a/src/transformers/modeling_utils.py b/src/transformers/modeling_utils.py
index 3f14d74979..f27ae1d884 100644
--- a/src/transformers/modeling_utils.py
+++ b/src/transformers/modeling_utils.py
@@ -982,7 +982,7 @@ class ModuleUtilsMixin:
                     torch.ones((batch_size, seq_length, prefix_seq_len), device=device, dtype=causal_mask.dtype),
                     causal_mask,
                 ],
-                axis=-1,
+                dim=-1,
             )
 
         extended_attention_mask = causal_mask[:, None, :, :] * attention_mask[:, None, None, :]
```

This is causing local typing check to fail.

**Comment by tarekziade** [comment]
run-slow: aimv2, aria, deepseek_vl_hybrid, eomt, eomt_dinov3, fast_vlm, flava, fuyu, idefics, imagegpt, kimi_k25, layoutlmv2, lfm2_vl, llava, minimax_m3_vl, pix2struct

**Comment by tarekziade** [comment]
Sorry I just realized that work was started earlier at https://github.com/huggingface/transformers/pull/47218 
I messed up when I did the triaging, sorry about that
